src/modules/repository/candlestick_repository.js

```diff
@@ -58,6 +58,15 @@ module.exports = class CandlestickRepository {
     });
   }
 
+  getCandlePeriods(exchange, symbol) {
+    return new Promise(resolve => {
+      const stmt = this.db.prepare(
+        `SELECT DISTINCT period from candlesticks where exchange = ? AND symbol = ? ORDER BY period ASC`
```

Comment:
this line breaks my usage. the result set to high its simply doing a full table search ;)

quick idea
 - use group
 - add filter
 - remove "n+1" (query per symbol) with an overall group

Reply:
@egisz i just needed to disable this; i also introduced some filter and grouping but still the pages running into timeout. currently i have no suitable idea to fetch the all values at once

Reply:
I pushed branch with time filter, but I see you already done this.
Could you check if you have indexes on candlestics table?
If you are unable to fetch candles of 5 days for one pair, how are you able to run backtests anyway?

Reply:
In a meanwhile, could you at least add "30m" period, it was why I started digging into this - I could not backtest 30m period :)

Reply:
> 5 days for one pair, how are you able to run backtests anyway

its because of the result set, eg already "1m" can have alot of items in 5 days. I guess maybe its better to use the symbol / instance config as a source. As its the input for the database anyways ;)

> 30m

not all exchanges have intervals like "30m", you can use "resample" it in the strategy based on "15m". I normalized exchanges that "15m" is supported everywhere